src/Psl/Filesystem/write_file.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Filesystem;
+
+use Psl;
+use Psl\Internal;
+use Psl\Str;
+
+use function file_put_contents;
+
+use const FILE_APPEND;
+use const LOCK_EX;
+
+/**
+ * Write $content to $file.
+ *
+ * @param bool $append If true, and $file exists, append $content to $file instead of overwriting it.
+ *
+ * @throws Psl\Exception\InvariantViolationException If the file specified by
+ *  $file does not exist, or is not writeable.
+ * @throws Exception\RuntimeException In case of an error.
+ */
+function write_file(string $file, string $content, bool $append = false): void
```

Comment:
Don't like the `$append` flag.
Can we split the function into:

- `write_file($file, $content)`
- `append_file($file, $content)`

Reply:
i agree on this, will move this internally, and have both `write_file` and `append_file` call it with the flag instead.